daemon-scheduler/pkg/deployment/deployment.go

```diff
@@ -21,18 +21,28 @@ import (
 	"github.com/pkg/errors"
 )
 
+// TODO: refactor to remove multiple environment retrievals from the database in methods like CreateSubDeployment
+// (check unit tests for GetEnvironment.Times(2 or 3))
```

Comment:
since the methods in this implementation of deployment call each other, ie CreateSubDeployment calls GetCurrentDeployment, in some instances we end up retrieving the environment multiple times which is inefficient. However, it might be required in the transactional code to make sure we have the latest environment when making changes. We should revisit this when adding transactions.